orc8r/protos/digest.proto

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package magma.orc8r;
+option go_package = "magma/orc8r/lib/go/protos";
+
+// DigestTree contains the full set of digest information for a particular network.
+// DigestTree is similar to a depth=2 Merkle tree.
+message DigestTree {
+  // root_digest is the amalgum of all leaf digests.
+  Digest root_digest = 1;
+  // leaf_digests contains per-object digests, along with the object IDs, sorted by ID.
+  repeated LeafDigest leaf_digests = 2;
+}
+
+// Digest contains the digest (hash) of some object.
+message Digest {
+  // md5_base64_digest is a base64-encoded MD5 digest.
+  string md5_base64_digest = 1;
+}
+
+message LeafDigest {
+  // id is the network-wide unique identifier of the object.
+  string id = 1;
+  // digest is the deterministic digest of the object.
+  Digest digest = 2;
+}
+
+// LeafDigestsToSerialize is used to serialize a list of leaf digests en masse.
+message LeafDigestsToSerialize {
```

Comment:
My concern with replacing `repeated LeafDigest` in other messages with `LeafDigests` was that it will result in an extra layer of data abstraction, and we'd have to unpack an extra data object to get to the list of leaf digests. Given that, not sure if I should choose consistency (using `LeafDigests` everywhere) over simplicity (directly use `repeated LeafDigest` in `DigestTree`)? @hcgatewood

Reply:
Hm. You're right that it does feel gross using a Message when we really want a slice. Okay I'm fine with leaving the others as slices (`repeated`), but still let's make this one `LeafDigests` and add a note that we're using a separate msg with the intention of it being for serialization

Rationale: I don't like naming things with intent -- the name should describe "what it is" not "how it's used". But adding a comment on "why" is totally fine